### DISCUSSION TITLE
[FR] Add reverse DNS lookup option for "pivpn -c"

### DISCUSSION THREAD

**Discussion by devZer0** [pivpn/pivpn#1398] [Feature requests]
When watching connected clients, it would be very useful to have an additional option which enables us to see the reverse dns name for the client's ip.

this way we would not only be able to easily and better see, from where they are connected - we could also see much more easily if there is mis-use of the vpn credentials (stolen identity, connection from china or elsewhere...)

i would be easier to detect if there are employees who are outsourcing their work to overseas :D  ( see https://www.theguardian.com/world/2013/jan/16/software-developer-outsources-own-job )

**Comment by orazioedoardo** [comment]
I tried some cellular and landline reverse lookups with `dig -x IPADDRESS` but I don’t always get an output (sometimes I get NXDOMAIN). Maybe a better way would be to get the autonomous system number (ASN) via whois and then ISP. What would you like to see for example?

**Comment by devZer0** [comment]
i think whois query could make sense, but it's more difficult to implement, as repsonse has multiple fields...

**Comment by coolapso** [comment]
Unfortunatelly PTR is not something that is very well implemented globally even big telecom companies forget to set them um properly. 

Also i have some concerns about this kind of feature ... this should be able to be disabled/enabled depending on how we wish to have our setup ... 

One might want a pivpn setup with 0 knowledge about the clients whereas someone else might want to have full control over it. 

However... i good solution might be this: 

```
┌─[f0rs3ti@deathstar]─[~]
└──╼ $curl ipinfo.io/88.198.194.195
{
  "ip": "88.198.194.195",
  "hostname": "docker1.4s3ti.net",
  "city": "Gunzenhausen",
  "region": "Bavaria",
  "country": "DE",
  "loc": "49.1166,10.7597",
  "org": "AS24940 Hetzner Online GmbH",
  "postal": "91710",
  "timezone": "Europe/Berlin",
  "readme": "https://ipinfo.io/missingauth"

```

**Comment by devZer0** [comment]
this looks cool, didn't know about ipinfo.io

the question is how to add to line-based output - maybe csv and add selectable fields to the last columns...

**Comment by orazioedoardo** [comment]
@devZer0. Well, it's JSON, you can use `jq` to parse it:

```
$ curl -s https://ipinfo.io/88.198.194.195 | jq -r '."org"'
AS24940 Hetzner Online GmbH
```

**Comment by devZer0** [comment]
i know, but to have it integrated in pivpn -c , you need that output in one line

i think it's more difficult how to handle/organize so it looks good in output then to handle it technically...

**Comment by orazioedoardo** [comment]
> i think it's more difficult how to handle/organize so it looks good in output then to handle it technically...

New column?
```
$ pivpn -c
::: Connected Clients List :::
Name      Remote IP            Virtual IP      Bytes Received      Bytes Sent      Last Seen                   Provider
user      127.0.0.1:50409      10.6.0.2        148B                188B            Feb 14 2020 - 16:23:09      AS24940 Hetzner Online GmbH
```

Also, this sends client IP addresses to ipinfo.io, what if the user doesn't like that?

**Comment by devZer0** [comment]
yes, new column(s), maybe provider would be enough - think we need to test how that looks with "real world ip's"

but make it an additional option, as this slows down output and sends data to 3rd party

**Comment by coolapso** [comment]
instead of provider i guess it would be better the region/city or is actually the provider you want?

**Comment by coolapso** [comment]
> Also, this sends client IP addresses to ipinfo.io, what if the user doesn't like that?

Very true.

**Comment by devZer0** [comment]
i think "-iploc fieldopts.....", e.g. "-iploc hostname,city,org"  is what i want..., so let the user decide what he wants to get displayed...
